### PR TITLE
new version scotch@6.0.5a

### PR DESCRIPTION
**Pull request by oliver-pola**
Updated scotch to release from 2018-02-15, included patch described at https://gforge.inria.fr/tracker/?func=detail&atid=1082&aid=19521&group_id=248

### CODE REVIEW

**Comment by healther** [comment]
You mention that you added a patch file, but that doesn't appear in the commit

**Comment by oliver-pola** [comment]
@healther no, the maintainer included that mentioned patch in the newer release which I just added.

**Comment by healther** [comment]
Should these patches be added to the older versions?

**Comment by oliver-pola** [comment]
Maybe, if someone faces the mentioned problem and needs to stay on scotch@6.0.4, that patch would be helpful. I just did that before I realized the new release, so I could add that also.

**Comment by healther** [comment]
I only skimmed the link you posted, but if I understood it correctly, than every `metis@6.0.0:6.0.4.999` will be affected right? But it seems like that would only be required for `scotch`, if that is the case (i.e. `metis` is fine for other users) than `scotch` should be responsible for patching `metis` (and I would have to look up how to do this)

**Comment by oliver-pola** [comment]
Yes that problem only occurs if you use scotch and metis together. I don't know about versions before 6.0.4.

**Comment by healther** [comment]
Can you add a `conflicts('metis@a:b')` statement for appropriate `a` and `b` to the `scotch` package with a comment linking to this issue? Otherwise lgtm to me

**Comment by oliver-pola** [comment]
I added that patch for scotch@6.0.4

File: var/spack/repos/builtin/packages/scotch/package.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ class Scotch(Package):
     url      = "http://gforge.inria.fr/frs/download.php/latestfile/298/scotch_6.0.4.tar.gz"
     list_url = "http://gforge.inria.fr/frs/?group_id=248"
 
+    version('6.0.5a', '8430dff7175a1dfd5a3258e75260cf71')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/scotch/package.py]: Is this an alpha release? We don't normally add alpha releases to Spack, as they tend to be very unstable. At the very least, if you want to add an alpha release, you should mark the previous version as `preferred=True`.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/scotch/package.py]: Strange. Their downloads page also lists many previous versions with `a` and `b` in them:
https://gforge.inria.fr/frs/?group_id=248

Perhaps it was just a small patch fix of 6.0.5 that was released soon after.

**Pull request merged**